### ISSUE TITLE
[Feature] Distinct list of urls for multimap matching

### ISSUE THREAD

**Issue by ercpe** [rspamd/rspamd#3003]
## Summary

It would be nice to have a distinct list of urls from emails to match against in the `multimap` module.

## Motivation

The `url` map type in the `multimap` module matches against every url in the message even if the very same url is repeated multiple times. This can lead to very high scores (since the score is multiplied by the number of urls that match the rule) even for low scores. For example an email that lists the same url twice in the text and html part gets four times the score of the rule.

## Describe alternatives you've considered

Since i couldn't find a transformation hook for the map type itself, i would like to see a new subtype of the `url` map type (e.g. `url:unique`) which produces a list of unique urls in the message.

**Comment by vstakhov** [comment]
In theory, you can use selector map + unique function for this purpose. Probably, there should also be `extract_specific_urls` as well to allow limiting of the number of URL queries. Also, all symbols in Rspamd could have two options (in groups.conf): `one_shot` meaning that one insertion is allowed or a more generic `nshots` option that can limit number of symbol's insertions to some value.

**Comment by ercpe** [comment]
> In theory, you can use selector map + unique function for this purpose. 

Do you mean sth. like 
```    type = "selector";
    selector = "urls.uniq"```
?

Unfortunately, `uniq` is 2.0+

> Also, all symbols in Rspamd could have two options (in groups.conf): `one_shot` meaning that one insertion is allowed or a more generic `nshots` option that can limit number of symbol's insertions to some value.

Yeah, my current approach is to balance the score within a group with `weight` and `max_score`, but 
 `one_shot` limits the total number of insertions per message, not per evaluation. But I guess underscoring is better than overscoring.

**Comment by vstakhov** [comment]
It is implemented in 2.0, closing issue.
